File: examples/py/minimal.py
import asyncio

from maestro_sdk.client import MaestroClient

BASE_URL = "http://localhost:8080"  # Your Maestro API URL
TOKEN = "your_auth_token"  # Your authentication token


async def main():
    client = MaestroClient(base_url=BASE_URL, token=TOKEN)

    try:
        print("Listing runs...")
        runs = await client.runs_list()  # Using the low-level generated client method
        if runs:
            for run in runs:
                print(f"Run ID: {run['id']}, Status: {run['status']}")
        else:
            print("No runs found.")

        # Start a new run
        print("\nStarting a new run...")
        new_run = await client.start_run(pipeline_id="my-python-pipeline", estimated_cost=0.02)
        print(f"New Run ID: {new_run['id']}")

        # Tail run logs
        print("\nTailing logs for the new run (simplified)...")
        logs = await client.tail_logs(new_run["id"])
        print("Logs:", logs)

        # Fetch evidence (placeholder)
        print("\nFetching evidence (conceptual)...")
        # evidence = await client.fetch_evidence("evidence-id-456")
        # print("Evidence:", evidence)

    except Exception as e:
        print(f"Error in SDK example: {e}")


if __name__ == "__main__":
    asyncio.run(main())

File: examples/ts/minimal.ts
import { createClient } from '../../sdk/typescript/src/client';

const BASE_URL = 'http://localhost:8080'; // Replace with your Maestro API URL
const TOKEN = 'your_token_here'; // Replace with your actual token or leave undefined

async function main() {
  const client = createClient(BASE_URL, TOKEN);

  try {
    console.log('Listing runs...');
    const runs = await client.listRuns();
    console.log('Runs:', runs.data); // Assuming runs.data contains the array of runs

    // Start a new run
    console.log('\nStarting a new run...');
    const newRun = await client.startRun('my-pipeline-id', {
      estimatedCost: 0.05,
    }); // Assuming startRun takes pipelineId and params
    console.log('New Run ID:', newRun.id);

    // Tail run logs
    console.log('\nTailing logs for the new run (simplified)...');
    const logs = await client.tailRunLogs(newRun.id); // Assuming tailRunLogs takes runId
    console.log('Logs:', logs); // Simplified: logs might be a string or stream

    // Fetch evidence (placeholder)
    console.log('\nFetching evidence (conceptual)...');
    // const evidence = await client.fetchEvidence("evidence-id-123"); // Assuming fetchEvidence is available
    // console.log("Evidence:", evidence);
  } catch (error) {
    console.error('Error in SDK example:', error);
  }
}

main();
